src/traces/carpet/plot.js

```diff
@@ -213,16 +213,16 @@ function drawAxisTitle(gd, layer, trace, t, xy, dxy, axis, xa, ya, labelOrientat
         }
 
         // In addition to the size of the labels, add on some extra padding:
-        var titleSize = axis.titlefont.size;
-        offset += titleSize + axis.titleoffset;
+        var titleSize = axis.title.font.size;
+        offset += titleSize + axis.title.offset;
 
         var labelNorm = labelOrientation.angle + (labelOrientation.flip < 0 ? 180 : 0);
         var angleDiff = (labelNorm - orientation.angle + 450) % 360;
         var reverseTitle = angleDiff > 90 && angleDiff < 270;
 
         var el = d3.select(this);
 
-        el.text(axis.title || '')
+        el.text(axis.title.text)
```

Comment:
FYI: removed `|| ''` because an empty title wouldn't be part of the joined `data` anyways.